### PR TITLE
Split tests

### PR DESCRIPTION
**Pull request by eddie-c-davis**
## Technical Description

This PR splits the GTClang integration tests for the optimizer passes in the gtclang/test/integration-test folder into Dawn unit tests in dawn/test/unit-test/dawn/Optimizer/Passes. 

### Resolves / Enhances

This PR contributes to the Parallelizer (SIR -> IIR) and Optimizer (IIR -> IIR) portions of issue #625

### CODE REVIEW

**Comment by eddie-c-davis** [comment]
launch jenkins

**Comment by eddie-c-davis** [comment]
launch jenkins

File: dawn/src/dawn/Serialization/IIRSerializer.cpp [hunk 1]
```diff
@@ -742,6 +742,11 @@ IIRSerializer::deserializeImpl(const std::string& str, IIRSerializer::Format kin
 std::shared_ptr<iir::StencilInstantiation> IIRSerializer::deserialize(const std::string& file,
                                                                       OptimizerContext* context,
                                                                       IIRSerializer::Format kind) {
+  return deserialize(file, kind);
+}
+
+std::shared_ptr<iir::StencilInstantiation> IIRSerializer::deserialize(const std::string& file,
```
Comment (twicki) [inline dawn/src/dawn/Serialization/IIRSerializer.cpp]: what is the difference to the `impl` here?

File: dawn/test/unit-test/dawn/Optimizer/Passes/input/DemoteTest01.sir [hunk 2]
```diff
@@ -0,0 +1,226 @@
+{
```
[the remark this answers is not in the document]
Comment (eddie-c-davis) [reply dawn/test/unit-test/dawn/Optimizer/Passes/input/DemoteTest01.sir]: * The IIRBuilder interface is challenging to use (personal preference)
* The IIR seems to be in flux so IIR files needed to be consistently updated
* It is unclear which passes have been applied to a particular IIR file
* The SIR seems to change less often
* Deserializing SIR and running the prerequisite passes to transform the IIR helps to understand dependencies between passes

These can be changed to use IIR if you like.

File: dawn/test/unit-test/dawn/Optimizer/Passes/TestPassSetCaches.cpp [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+               const std::vector<std::vector<std::string>>& fieldNames,
+               const std::vector<std::vector<iir::Cache::CacheType>>& cacheTypes,
+               const std::vector<std::vector<iir::Cache::IOPolicy>>& ioPolicies) {
+    if(nStencils < 1)
+      nStencils = 1;
+
+    std::shared_ptr<iir::StencilInstantiation> instantiation =
+        CompilerUtil::load(filename, options_, context_, TestEnvironment::path_);
+
+    ASSERT_TRUE(CompilerUtil::runGroup(PassGroup::Parallel, context_, instantiation));
```
[the remark this answers is not in the document]
Comment (eddie-c-davis) [reply dawn/test/unit-test/dawn/Optimizer/Passes/TestPassSetCaches.cpp]: Goal was to maintain consistency with groups in optimizer-refactor branch, these can be changed to individual passes if desired

**Comment by eddie-c-davis** [comment]
launch jenkins

**Comment by eddie-c-davis** [comment]
launch jenkins

**Comment by jdahm** [comment]
Assuming the original tests are deleted in a follow-up PR, I think this is fantastic!

**Pull request merged**